pkgs/top-level/haskell-ng.nix

```diff
@@ -16,7 +17,10 @@ rec {
     ghc722 = callPackage ../development/compilers/ghc/7.2.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc742 = callPackage ../development/compilers/ghc/7.4.2.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
     ghc763 = callPackage ../development/compilers/ghc/7.6.3.nix { ghc = compiler.ghc704Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
-    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix { ghc = compiler.ghc742Binary; gmp = pkgs.gmp.override { withStatic = true; }; };
+    ghc784 = callPackage ../development/compilers/ghc/7.8.4.nix {
+      ghc = if pkgs.stdenv.isDarwin then compiler.ghc783Binary else compiler.ghc742Binary;
```

Comment:
I worried about using the `ghc783Binary` expression because it is in such a bad shape, really. I don't think it builds anywhere except `x86_64-darwin`; it certainly doesn't build on Linux. Are you sure that the newer compiler is necessary to boostrap 7.8.4? To me, it's not intuitively clear why the older compiler can bootstrap GHC on Linux but not on Darwin. Is there any particular reason for that?